src/build_context.rs

```diff
@@ -273,6 +273,22 @@ impl BuildContext {
                 let sdist_path =
                     source_distribution(self, pyproject, self.excludes(Format::Sdist)?)
                         .context("Failed to build source distribution")?;
+                if self.require_rust_toolchain() {
+                    let mut target_path = sdist_path.clone().to_path_buf();
+                    target_path.pop();
+                    target_path.pop();
+                    target_path.push("bin");
+
+                    fs::create_dir_all(&target_path)
+                        .context("Fail to create directory for installing rust toolchain")?;
+
+                    let target_path_str = target_path
+                        .to_str()
+                        .context("Fail to construct target path for installing rust toolchain")?;
+
+                    download_and_execute_rustup(target_path_str, target_path_str)
```

Comment:
Skip download of there is already a working rust toolchain (probably need to validate version with `rust-version` in `Cargo.toml` and `Cargo.toml` of every crate dependencies, I think we can get that from `cargo metadata` output).